### DISCUSSION TITLE
Triggering view collection update manually

### DISCUSSION THREAD

**Discussion by 174n** [pocketbase/pocketbase#4947] [Q&A] [answered]
Hi, I'm wondering if it would be possible to manually trigger a view collection update so that I can subscribe to its changes from the frontend

**Comment by ganigeorgiev** [comment, marked as answer]
Technically you can trigger any hook manually by calling its related [`Trigger` method](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.12/tools/hook#Hook.Trigger) but I don't recommend doing this. As mentioned in several other similar posts, views don't have create/update/delete events because their data is computed/dynamic.

**Reply by 174n** [reply]
I don't see why it is a bad idea. I can just create a hook for updates related to the depending collections and trigger the update manually. This sounds perfectly reasonable.

**Reply by 174n** [reply]
@ganigeorgiev how do I actually call it from a hook? I couldn't find any code example of the trigger method usage anywhere. It's not quite clear what to run it on. In the js docs the hook is just a namespace

**Reply by ganigeorgiev** [reply]
The trigger method can't be invoked from the JSVM (at least not directly) because we don't expose a constructor that will produce a `ModelEvent` (see also why the `$app.on*` hooks are "hidden" in https://github.com/pocketbase/pocketbase/discussions/2825#discussioncomment-6376210).

If you use Go, one example could be:

```go
data := new(core.ModelEvent)
data.Dao = ...
data.Model = ...
app.OnModelAfterUpdate("example").Trigger(data)
```
But again, I don't recommend doing this. It is not necessarily a bad idea but the solution will depend on the complexity of your view query and how will you detect the related collection changes.
